lib/ssz.ex

```diff
@@ -139,4 +154,305 @@ defmodule Ssz do
 
   @spec decode_u256(binary) :: non_neg_integer
   def decode_u256(num), do: :binary.decode_unsigned(num, :little)
+
+  ##### elixir native ssz
+  @bytes_per_length_offset 4
+  @bytes_per_chunk 32
+  @bits_per_byte 8
+
+  ### ENCODE ######
+  def serialize(%struct{} = map) when is_struct(map) do
```

Comment:
`%struct{} = map` already checks if `map` is a struct, so `is_struct` shouldn't be needed.